src/app/auth/permission.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+export type Permission = {
+  exp: number,
+  resource_set_id: string,
+  resource_set_name: string,
+  scopes: Array<string>
+};
+
+@Injectable()
+export class PermissionService {
+  jwtHelper: JwtHelperService;
+  constructor() {
+    this.jwtHelper = new JwtHelperService();
+  }
+
+  /**
+   * Returns all the scopes a user has for a specific resource.
+   * @param resourceId ID of a specific resource such as a Space
+   */
+  getAllScopes(resourceId: string): Array<string> {
+    const permissions = this.getPermission(resourceId);
+    return permissions ? permissions.scopes : [];
+  }
+
+  /**
+   * Checks if a user has a specific scope for a resource.
+   * @param resourceId ID of a specific resource such as a Space
+   * @param scope the scope you want to check for. Ex - `can edit`
+   */
+  checkScope(resourceId: string, scope: string): boolean {
```

Comment:
A better name for this API might be `isResourceInScope` or `isInScope` to align with `is` functions returning a boolean.

Reply:
This method will be used to check if a user has a specific scope for a resource or not. That way the names like `isResourceInScope` or `isInScope` doesn't make sense here. Something like `hasScopeForResource` or `hasScope` might be more meaningful here.

Reply:
Thanks for the clarification. I'm ok with `hasScope`.